tests/test_particular_profiles.py

```diff
@@ -0,0 +1,267 @@
+        """
+        test error raising in case of missing profile factors
+
+        """
+        profile_factors = {
+            "week": {"day": 0.6, "night": 0.5},
+        }
+        msg = "Missing entry for 'weekend' in profile_factors."
+        with pytest.raises(ValueError, match=msg):
+            self.ilp.simple_profile(
+                1.0,
+                profile_factors=profile_factors,
+            )
+
+        profile_factors = {
+            "week": {"day": 0.6},
+        }
+        msg = "Missing entry for 'night' in profile_factors for 'week'."
+        with pytest.raises(ValueError, match=msg):
+            self.ilp.simple_profile(
+                1.0,
+                profile_factors=profile_factors,
+            )
+
+    def test_simple_profile_holiday_is_sunday(self):
+        """
+        test with default values and holiday_is_sunday = True
+
+        """
+        profile_factors = {
+            "week": {"day": 0.6, "night": 0.5},
+            "weekend": {"day": 0.7, "night": 0.4},
+            "holiday": {"day": 0.8, "night": 0.3},
+        }
+        df = self.ilp_holiday.simple_profile(
+            1.0, profile_factors=profile_factors
+        )
+
+        # check factors in self.dataframe["ind"]
+        # holiday - now treated as Sunday
+        assert (
+            self.ilp_holiday.dataframe.at[
+                pd.Timestamp("2010-01-01 6:00"), "ind"
+            ]
+            == 0.4
+        )
+        assert (
+            self.ilp_holiday.dataframe.at[
+                pd.Timestamp("2010-01-01 7:00"), "ind"
+            ]
+            == 0.7
+        )
+        assert (
+            self.ilp_holiday.dataframe.at[
+                pd.Timestamp("2010-01-01 23:00"), "ind"
+            ]
+            == 0.7
+        )
+        # weekend day
+        assert (
+            self.ilp_holiday.dataframe.at[
+                pd.Timestamp("2010-01-02 6:00"), "ind"
+            ]
+            == 0.4
+        )
+        assert (
+            self.ilp_holiday.dataframe.at[
+                pd.Timestamp("2010-01-02 7:00"), "ind"
+            ]
+            == 0.7
+        )
+        assert (
+            self.ilp_holiday.dataframe.at[
+                pd.Timestamp("2010-01-02 23:00"), "ind"
+            ]
+            == 0.7
+        )
+        # week day
+        assert (
+            self.ilp_holiday.dataframe.at[
+                pd.Timestamp("2010-01-04 6:00"), "ind"
+            ]
+            == 0.5
+        )
+        assert (
+            self.ilp_holiday.dataframe.at[
+                pd.Timestamp("2010-01-04 7:00"), "ind"
+            ]
+            == 0.6
+        )
+        assert (
+            self.ilp_holiday.dataframe.at[
+                pd.Timestamp("2010-01-04 23:00"), "ind"
+            ]
+            == 0.6
+        )
+
+        # check sum of factors to make sure not only the tested factors are
+        # correct
+        assert np.isclose(self.ilp_holiday.dataframe["ind"].sum(), 98.9)
```

Comment:
Why 98.9?

Reply:
That's just what all factors add up when they are set correctly. Above I check for certain time steps and here I want to indirectly check all time steps.

Reply:
For me, this looks like "it should add up to 100 but there are rounding errors". Maybe, add a comment why this is the expected result.

Reply:
I elaborated the comment. I hope, this makes it clearer now?

Reply:
Prefect, thanks. (When values are in the range of 100, people tend to assume things.)